src/util/neuralNetwork.py

```diff
@@ -0,0 +1,142 @@
+import pandas as pd
+import numpy as np
+import logging.config
+import sys
+import random
+from src.datamodel.Column import DataDictionary as dd
+import tensorflow as tf
+from tensorflow import keras
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import roc_auc_score
+from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score
+from sklearn.model_selection import GridSearchCV
+import matplotlib.pyplot as plt
+import datetime
+import src.util.FileProvider as FP
+logger = logging.getLogger(__name__)
+
+def nnModel(data: pd.DataFrame, parameters: list, target: list) -> pd.DataFrame:
+    numeric_columns = data.select_dtypes(include=['object']).columns
+
+    # Set random seeds for TensorFlow and NumPy
+    tf.random.set_seed(42)
+    np.random.seed(42)
+    random.seed(42)
+# Convert selected columns to numeric
+    data[numeric_columns] = data[numeric_columns].apply(pd.to_numeric, errors='coerce')
+
+    X = data[parameters]
+    y = data[target]
+    y = y.values.ravel()
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
+    logger.debug(f'data: {data.info()}')
+    if not X.empty:
+        # Sample data generation
+        # Define hyperparameters to search
+        param_grid = {
+            'batch_size': [32, 64, 128],
+            'epochs': [50, 100, 200]
+        }
+
+        # Create the neural network model as a function
+        #neural_network = tf.keras.wrappers.scikit_learn.KerasClassifier(build_fn=create_model, input_dim=X_train.shape[1], verbose=0)
+
+        # Perform grid search
+        #model = GridSearchCV(estimator=neural_network, param_grid=param_grid, cv=3)
+        #grid_result = model.fit(X_train, y_train)
+        # Create the model
+        input_dim = X_train.shape[1]
+        model = create_model(input_dim)
+        # Train the model with early stopping based on validation loss
+        early_stopping = tf.keras.callbacks.EarlyStopping(monitor='val_loss', patience=5, restore_best_weights=True)
+        history = model.fit(X_train, y_train, validation_data=(X_test, y_test), epochs=100, batch_size=64, callbacks=[early_stopping])
```

Comment:
Let's split this function into 3 different steps. First one is the model training. X and Y would be parameters and it should return history and model